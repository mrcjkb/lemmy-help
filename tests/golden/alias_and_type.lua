local U = {}

---@alias NoDesc string

---All the lines in the buffer
---@alias Lines string[]

---Vim operator-mode motions.
---
---Read `:h map-operator`
---@alias VMode
---| '"line"' # Vertical motion
---| '"char"' # Horizontal motion
---| 'v'
---| `some.ident` # Some identifier

---Returns all the content of the buffer
---@return Lines
function U.get_all()
    return vim.api.nvim_buf_get_lines(0, 0, -1, false)
end

---List of all the lines in the buffer
---It can be more than one
---@type Lines lines in a buffer
---@see Lines
U.LINES = {}

---Global vim mode
---@type VMode
---@usage `print(require('plugin').VMODE)`
U.VMODE = 'line'

return U
