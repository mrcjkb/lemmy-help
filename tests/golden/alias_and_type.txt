NoDesc                                                                  *NoDesc*

    Type: ~
        string


Lines                                                                    *Lines*
    All the lines in the buffer

    Type: ~
        string[]


VMode                                                                    *VMode*
    Vim operator-mode motions.

    Read `:h map-operator`

    Variants: ~
        ("line")      Vertical motion
        ("char")      Horizontal motion
        ("v")
        (some.ident)  Some identifier


U.get_all()                                                          *U.get_all*
    Returns all the content of the buffer

    Returns: ~
        (Lines)


U.LINES                                                                *U.LINES*
    List of all the lines in the buffer
    It can be more than one

    Type: ~
        (Lines)  lines in a buffer

    See: ~
        |Lines|


U.VMODE                                                                *U.VMODE*
    Global vim mode

    Type: ~
        (VMode)

    Usage: ~
>lua
        print(require('plugin').VMODE)
<



